### PR TITLE
Supporting accent characters in Search

### LINKED ISSUE #9: Support accent characters in search
Originaly opened by @fredrocious

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're addressing an (ancient) issue: we're turning the Search Predicate into Diacritic Insensitive.

This means that: from now on, Search will be "accent insensitive". For instance, if a note contains the word `esdrújula`, and the keyword is `esdrujula` (that is: lacks the accent), it's expected to show up in the search results.

Closes #9

Thank you @bummytime !!

### Test
1. Launch the app
2. Add a new note with the word `esdrújula` and the tag `jamás`
3. Press back

- [x] Verify that the word `esdrujula` (without accent) yields the note you've just added
- [x] Verify that the word `jamas` yields the tag `jamás` (with accent)

### Release
`RELEASE-NOTES.txt` was updated in f377d28 with:

> Search is now diacritic insensitive: type search terms with or without accents to get matches!

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 -   Search results now display matching tags!
 -   We're now supporting the `tag:keyword` search operator. Simplenote will yield Notes whose tag names contain such *keyword*.
 -   Notes List's Interface is now looking more beautiful than ever.
+-   Search is now diacritic insensitive: no need to type keywords with accents to get matches!
```
Comment (bummytime) [inline RELEASE-NOTES.txt]:
```suggestion
-   Search is now diacritic insensitive: type search terms with or without accents to get matches!
```

@jleandroperez should we change this text to something like this ☝️ ?

File: Simplenote/Classes/NSPredicate+Simplenote.swift [hunk 2]
```diff
@@ -14,15 +14,15 @@ extension NSPredicate {
...
+                output.append( NSPredicate(format: "content CONTAINS[cd] %@", keyword) )
                 continue
             }
 
             guard !tag.isEmpty else {
                 continue
             }
 
-            output.append( NSPredicate(format: "tags CONTAINS[c] %@", tag) )
+            output.append( NSPredicate(format: "tags CONTAINS[cd] %@", tag) )
```
Comment (bummytime) [inline Simplenote/Classes/NSPredicate+Simplenote.swift]: Such a tiny fix for a long-stand bug! 😆

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (jleandroperez) [reply RELEASE-NOTES.txt]: Thank you !!! Patching up as suggested 🔥

File: Simplenote/Classes/NSPredicate+Simplenote.swift (hunk 2 above)
Comment (jleandroperez) [reply Simplenote/Classes/NSPredicate+Simplenote.swift]: I know right!!

**Comment by jleandroperez** [comment]
Thank you Bummy!

**Pull request merged**